js/src/behaviors/dismiss.ts

```diff
@@ -0,0 +1,40 @@
+export function initDismiss() {
+	window.addEventListener(
+		'click',
+		({ target }: Event) => {
+			const dismissButton = target instanceof Element && target.closest('[data-dismiss]');
+
+			if (!dismissButton) {
+				return;
+			}
+
+			const dismissTarget = dismissButton.closest('[data-dismissable]');
+
+			if (!dismissTarget) {
+				return;
+			}
+
+			dismissElement(dismissTarget);
+		},
+		{ passive: true }
+	);
+}
+
+async function dismissElement(dismissTarget: Element) {
+	if (dismissTarget.hasAttribute('disappearing')) {
+		dismissTarget.classList.add('is-dismissed');
+
+		const disappearing = dismissTarget.getAttribute('disappearing');
```

Comment:
Something to noodle over, but do we really need/want these animation in our CSS? If the answer is no or maybe not, we might also be able to leverage the JS only API [element.animate](https://developer.mozilla.org/en-US/docs/Web/API/Element/animate), which would allow us to do the same. I don't feel strongly about this, feel free to resolve